source/opt/interface_var_sroa.cpp

```diff
@@ -134,43 +134,34 @@ SpvStorageClass GetStorageClass(Instruction* var) {
 
 }  // namespace
 
+bool InterfaceVariableScalarReplacement::HasExtraArrayness(Instruction* var) {
```

Comment:
Are you sure this is correct?  In our internal discussion Alan said

> So for tess eval, outputs without patch are arrayed and for tess control input and output without patch are arrayed.

However, I do not see anything checking the shader stage or whether or not it is an input or output.  This is a sign that your testing is too simple.  You only have 1 test.  You need a lot more test to check all of the cases.

Reply:
Sure. I will add more tests. I still focus on the refactoring now. Sorry for keep updating it directly. I will update this part now.

Reply:
done